Automattic/jetpack
# Extensions: auto-generate public path

## Proposed changes:

Adding multiple instances of certain blocks (e.g., the Tiled Gallery) in the editor causes the `Jetpack_Block_Assets_Base_Url` variable to be set several times. Simply put, this is because the `render` function server-side is called multiple times, which in turn causes [load_scripts_as_required](https://github.com/Automattic/jetpack/blob/b3f173e91846698a6f0d0933bb20f0b61015e330/projects/plugins/jetpack/class.jetpack-gutenberg.php#L541C25-L541C49) to be called  and `Jetpack_Block_Assets_Base_Url` to be set as many times (see the link in the _Jetpack product discussion_ for a more detailed discussion)

With [Webpack's automatic publicPath](https://webpack.js.org/guides/public-path/#automatic-publicpath) feature, we don't need to set the public path explicitly anymore, and we can get rid of `Jetpack_Block_Assets_Base_Url`. This PR takes advantage of this feature and removes pieces of code that are now deprecated. This Webpack feature relies on `document.currentScript`, which I believe has a good enough [browser support](https://caniuse.com/document-currentscript).

_Note: this issue had no impact on the user experience._

### Other information:


## Jetpack product discussion


p1703837496108339-slack-CBG1CP4EN

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






- Spin up a test site with this branch
- Build the plugin: `jetpack build plugins/jetpack`
- Create a new post
- Add several instances of the Tiled Gallery block
<img width="660" alt="Screenshot 2024-01-09 at 2 16 17 PM" src="https://github.com/Automattic/jetpack/assets/1620183/859251e2-0af0-4246-a553-56a93b51ccb2">

- Open the preview
- Check everything works as expected and that there's no error in the console
- Back to the editor, add the Map block, and check it's not broken (it's the only block calling `loadBlockEditorAssets`, which we're updating in this PR)
